### PR TITLE
Cargo / passengers are no longer transported to player on save load

### PR DESCRIPTION
**Pull request by yjhn**
**Bugfix:** This PR addresses issue #1624.
This is a simplified and corrected version of #5070.

## Fix Details
This fix generates UUIDs for missions and then uses them to store in which ships mission cargo / passengers currently are.

## Testing Done
Same as in #5070.

## Save File
[John Brown.txt](https://github.com/endless-sky/endless-sky/files/5983468/John.Brown.txt)

Steps to verify:
1. load the game
2. the passenger mission does not complete until you depart, wait for escorts to arrive and land again (without this patch the mission will complete as soon as the save is loaded)

### CODE REVIEW

**Comment by tehhowch** [comment]
@tatami4 please redo this PR using the game's UUID implementation

**Comment by tatami4** [comment]
> @tatami4 please redo this PR using the game's UUID implementation

Done, it should work the same as before.

**Review by tehhowch — CHANGES_REQUESTED** [review]
An alternate approach to the one taken currently would be to save the mission's cargo distribution information in the mission itself, or perhaps in the player object. It's possible these approaches would improve things, but also possible that they are worse in some ways, too. Would you like to investigate alternative approaches? Separate branches would be fine.

One additional issue with the proposed solution is that it assumes mission cargo can only be on player ships. The game at one point allowed random ships to plunder passengers, and I'm sure someone would love to be able to create a mission where the player has to plunder an NPC and actually take an item. I think that may only be possible if we store the cargo mapping in each mission (and promote any random NPCs which steal mission cargo into mission-associated ships), but I could be wrong. (I'm definitely not asking for all of that to be added in this PR.)

File: source/PlayerInfo.cpp [hunk 1]
```diff
@@ -221,24 +221,23 @@ void PlayerInfo::Load(const string &path)
 		else if(child.Token(0) == "mission")
 		{
 			missions.emplace_back(child);
-			cargo.AddMissionCargo(&missions.back());
+			// For compatibility with old saves we load the mission the old way if it
+			// does not have UUID.
+			if(missions.back().UUID().IsEmpty())
+				cargo.AddMissionCargo(&missions.back());
```
[the remark this answers is not in the document]
Comment (tatami4) [reply source/PlayerInfo.cpp:227]: Yes, this isn't good, but I can't think of a way to store cargo and passenger distribution in a way that would be compatible with current saves. There is no way to infer cargo and passenger distribution from current save format.

**Comment by warp-core** [comment]
Was the possibility of recording the distribution of mission cargo/passengers by system or ship UUID in the mission considered?
That seems like it'd be better than this.

**Comment by Amazinite** [comment]
Closing in favor of #7505.

**Pull request closed without merging**